codalab/apps/api/views/competition_views.py

```diff
@@ -739,19 +740,17 @@ def get(self, request, *args, **kwargs):
                                     'status': sub.status.codename,
                                     'logs': log_sas_urls
                                 }
-                                response = Response(temp_data, status=status.HTTP_200_OK)
-                                return response
+                                return Response(temp_data, status=status.HTTP_200_OK)
                         except (KeyError, StopIteration):
                             pass
             except (KeyError, IndexError):
-                response = Response(
-                    "Submission is not on leaderboard or is not accessible!",
-                    status=status.HTTP_404_NOT_FOUND
-                )
+                pass
         except CompetitionSubmission.DoesNotExist:
-            # This is probably not the right response...
-            response = Response("Submission is not on leaderboard or is not accessible!", status=status.HTTP_404_NOT_FOUND)
-        return response
+            raise Http404("Submission is not on leaderboard or is not accessible!")
+        return Response(
```

Comment:
this is returning a 404 not raising it?

Reply:
👍 

I also tried to differentiate each case a bit, since this looks messy. I still think this API point needs some more clean-up. I'm not too sure on the best way to do that since there's many cases where something can go wrong retrieving the score from the submission.